pkg/onboarding/usecases/signup.go

```diff
@@ -498,5 +504,77 @@ func (s *SignUpUseCasesImpl) RemoveUserByPhoneNumber(ctx context.Context, phone
 		utils.RecordSpanError(span, err)
 		return exceptions.NormalizeMSISDNError(err)
 	}
-	return s.onboardingRepository.PurgeUserByPhoneNumber(ctx, *phoneNumber)
+	return s.repo.PurgeUserByPhoneNumber(ctx, *phoneNumber)
+}
+
+// RegisterUser creates a new userprofile
+func (s *SignUpUseCasesImpl) RegisterUser(ctx context.Context, input dto.RegisterUserInput) (*dto.RegisteredUserResponse, error) {
+	ctx, span := tracer.Start(ctx, "RegisterUser")
+	defer span.End()
+
+	phoneNumber, err := s.baseExt.NormalizeMSISDN(input.PhoneNumber)
+	if err != nil {
+		utils.RecordSpanError(span, err)
+		return nil, exceptions.NormalizeMSISDNError(err)
+	}
+
+	timestamp := time.Now().In(pubsubtools.TimeLocation)
+
+	userProfile := profileutils.UserProfile{
+		PrimaryEmailAddress: &input.Email,
+		UserBioData: profileutils.BioData{
+			FirstName:   &input.FirstName,
+			LastName:    &input.LastName,
+			Gender:      enumutils.Gender(input.Gender),
+			DateOfBirth: &input.DateOfBirth,
+		},
+		CreatedByID: &input.CreatedBy,
```

Comment:
Get Logged in user and use it's  profile ID as `createdBy`

Reply:
that's what I did, but how do u get logged in user with inter service authentication?

Reply:
Get the logged in user UID from the other service

Reply:
And that's why I'm passing that ID so as to pass it during ISC call.